clang/lib/CodeGen/CodeGenFunction.cpp

```diff
@@ -743,22 +743,10 @@ void CodeGenFunction::EmitOpenCLKernelMetadata(const FunctionDecl *FD,
 
   if (const SYCLIntelMaxWorkGroupSizeAttr *A =
           FD->getAttr<SYCLIntelMaxWorkGroupSizeAttr>()) {
-    ASTContext &ClangCtx = FD->getASTContext();
-    Optional<llvm::APSInt> XDimVal = A->getXDimVal(ClangCtx);
-    Optional<llvm::APSInt> YDimVal = A->getYDimVal(ClangCtx);
-    Optional<llvm::APSInt> ZDimVal = A->getZDimVal(ClangCtx);
-
-    // For a SYCLDevice SYCLIntelMaxWorkGroupSizeAttr arguments are reversed.
-    if (getLangOpts().SYCLIsDevice)
-      std::swap(XDimVal, ZDimVal);
-
```

Comment:
This seems to  be a bug with old implementation. Should we reverse the attribute arguments for "max_work_group_size"?

Reply:
>>Didn't we do this because we wanted reqd_work_group_size, max_work_group_size, and work_group_size_hint to all have the same argument order?

Yes, right based on discussion here: https://github.com/intel/llvm/issues/3361